src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs

```diff
@@ -111,8 +111,21 @@ private CompletionProvider GetCompletionProvider(CompletionItem item)
                     var mappedSpan = triggerSnapshotSpan.TranslateTo(
                         this.SubjectBuffer.CurrentSnapshot, SpanTrackingMode.EdgeInclusive);
 
+                    // If the client is providing a caret position, then it's likely they're
+                    // making a large/complex change to the document.  In this case try to
+                    // get the editor to actually apply this as a minimal-change so that the
+                    // document change looks nice and tidy.  This won't mess anything up as we
+                    // will then place the caret manually.
+                    //
+                    // If the caret position is not provided, then we do not want a minimal-change
```

Comment:
💭 The part I have the most trouble with here is it's hard for users working on a completion subsystem to understand the relevance/importance of `CompletionChange.NewPosition` to a subtle usability feature.

:question: Is it not possible to calculate the new position here (as simply the end of the inserted text) and then always use `EditOptions.DefaultMinimalChange`? This would preserve the current behavior for current use cases while reducing the likelihood that increasingly complex completion operations produce unexpected results. It also reduces the subtlety of unexpected behaviors: if the end of the inserted text is not the expected location of the caret then it's clear how to fix it, but it's far from obvious if the position's calculation impacts a feature like the change tracking margin.